crates/sui-bcs-json/README.md

```diff
@@ -0,0 +1,71 @@
+# Description
+
+This library provides an interface for transforming BCS data into JSON and vice-versa. The aim is to provide a way to support BCS in an interoperable manner for languages that do not have native BCS support, but have C interoperability.
+
+# Design choices
+
+The library was designed to be as simple as possible to allow users to convert to JSON or to BCS by passing the `typename` (as BCS requires to know before hand the type of the encoded data). Specifically, the types that are supported to be converted as of now are the `TransactionKind` and its subtypes from the main `Sui Transaction` type [transaction.rs](https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/transaction.rs)
+
+## API
+The library offers two functions for serializing and deserializing data:
+* **sui_bcs_from_json**
+```c
+size_t sui_bcs_from_json(const char *type_name,
```

Comment:
nit: I have a slight preference for calling this `sui_json_to_bcs` to make the distinction between this API and the next one more obvious.